suppression-list/csv2optout.py

```diff
@@ -0,0 +1,208 @@
+import csv
+import logging
+import re
+import sys
+from datetime import datetime, timezone
+from pathlib import Path
+
+import click
+
+logger = logging.getLogger(__name__)
+
+
+SQL_COMMANDS_PRE = """
+CREATE OR REPLACE PROCEDURE raise_notice (s TEXT) LANGUAGE plpgsql AS
+$$
+BEGIN
+    RAISE NOTICE '%', s;
+END;
+$$;
+
+CREATE TEMPORARY TABLE IF NOT EXISTS imported_{tmp_suffix} (
+  email TEXT,
+  tstxt TEXT,
+  unsubscribe_reason TEXT
+);
+
+CALL raise_notice('Start CSV import ({csv_rows_count} rows)...');
+
+COPY imported_{tmp_suffix}(email, tstxt, unsubscribe_reason)
+  FROM '{csv_path}' -- fullpath
+  DELIMITER '{delimiter}'
+  {headers}
+  QUOTE AS '"';
+
+CALL raise_notice('CSV import done.');
+
+CALL raise_notice('Join on existing contacts...');
+
+CREATE TABLE IF NOT EXISTS optouts_{tmp_suffix} (
```

Comment:
We could make this one a temp table too and let it drop at session close. That would save us from cleanup and remove the need for a temp suffix.

Reply:
I don't think so, the idea is that the update batches can be executed over the course of several days if necessary